src/ui/page/dataverse/dataset/dataset.tsx

```diff
@@ -91,15 +92,29 @@ const datasetMetadata: ItemGeneralMetadata[] = [
   }
 ]
 
-const Dataset = (): JSX.Element => {
+const Dataset: FC = () => {
   const { id } = useParams<string>()
-  const [dataset, setDataset] = useState<Option<DataverseItemDetails>>(none)
+  const [dataset, setDataset] = useState<O.Option<DataverseItemDetails>>(O.none)
+  const [isLoading, setIsLoading] = useState(true)
 
   useEffect(() => {
-    id && setDataset(getResourceDetails(id))
+    pipe(
+      id,
+      O.fromNullable,
+      O.map(getResourceDetails),
+      O.map(resourceDetails => {
```

Comment:
`map` is traditionally used for transformations rather than side effects, so I would not use it in this case. I believe that to perform a side effect, it would be more suitable to use an `IO`, and in this specific context where you have to deal with a value or not, an [IOOption](https://gcanti.github.io/fp-ts/modules/IOOption.ts.html).

As it is idiomatic, it would be nice to define this implementation as a reference implementation.

Reply:
@ccamel thanks for pointing this out. In order to define it as a reference, I would like to have your confirmation about IO in the React context.
`getResourceDetails` currently returns and Option, and probably a task once the vue is wired to the ontology.

Considering `getResourceDetails` is an Option, here is my proposal:
```tsx
const resourceDetails = useMemo(
    () => pipe(id, O.fromNullable, IOO.fromOption, IOO.map(getResourceDetails)),
    [id]
)

useEffect(() => {
  pipe(
    () => setIsLoading(true),
     resourceDetails,
     O.map(setDataspace),
     O.map(() => setIsLoading(false))
  )
}, [resourceDetails])
```

Do we consider state updates within useEffect as side effect as well?
If yes, which reflects probably better the FP principles, we probably want an implementation closer to this one:
```tsx
const resourceDetails = useMemo(
    () => pipe(id, O.fromNullable, IOO.fromOption, IOO.map(getResourceDetails)),
    [id]
)

useEffect(() => {
  pipe(
    IO.of(setIsLoading(true)),
    IO.map(resourceDetails),
    IO.flatMap(
      O.fold(
        () => IO.of(setIsLoading(false)),
        r => {
          setDataspace(r)
          return IO.of(setIsLoading(false))
        }
      )
    )
  )()
}, [resourceDetails])
```

What do you think? Thanks

Reply:
As we discussed today, we limit the usage of fp-ts to the domain, store, or utility functions. The bug fix has been updated accordingly.